### DISCUSSION TITLE
Is there any way to restart a dead device automatically

### DISCUSSION THREAD

**Discussion by JingtaiChen** [stm32duino/Arduino_Core_STM32#2169] [Q&A] [answered]
When I used Arduino Uno R:3 before, R3 could automatically restart when there was an unexpected situation in the program, but now STM32 will only crash and get stuck there, is there any way
Can STM32 be automatically restored to the beginning of the program after jamming to restart the operation?

**Comment by fpistm** [comment, marked as answer]
Hi @CJT1111 
you can have a look to this library: 
https://github.com/stm32duino/Arduino_Core_STM32/tree/main/libraries/IWatchdog

Or redefine this to call `NVIC_SystemReset();`
https://github.com/stm32duino/Arduino_Core_STM32/blob/586319c6c2cee268747c8826d93e84b26d1549fd/cores/arduino/stm32/stm32_def.h#L138-L142

Or redefine some exception handler example for a STM32G4:
https://github.com/stm32duino/Arduino_Core_STM32/blob/586319c6c2cee268747c8826d93e84b26d1549fd/system/Drivers/CMSIS/Device/ST/STM32G4xx/Source/Templates/gcc/startup_stm32g471xx.s#L253-L283

which fallback to this:
https://github.com/stm32duino/Arduino_Core_STM32/blob/586319c6c2cee268747c8826d93e84b26d1549fd/system/Drivers/CMSIS/Device/ST/STM32G4xx/Source/Templates/gcc/startup_stm32g471xx.s#L108-L120

**Comment by JingtaiChen** [comment]
Thanks, I will look at the first example and do a restart function in whil, but please do not close the problem, I will return to this post after experimenting.

**Reply by fpistm** [reply]
@CJT1111 
it seems you open several discussions around the same subject: #1909 and #2140.
Please avoid this, I close also this one as it is answered. You have several way to implement a restart.
